### ISSUE TITLE
exceptions.AssertionError: ('0.0.0.0', 7759)

### ISSUE THREAD

**Issue by synctext** [Tribler/tribler#1623]
```
tribler$ ./tribler.sh 
ERROR   1440706923.60             log:443   Unhandled Error
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/pouwelse/GIT-pulls/tribler/Tribler/Core/Utilities/twisted_thread.py", line 55, in _reactor_runner
    reactor.run(installSignalHandlers=False)
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1192, in run
    self.mainLoop()
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1201, in mainLoop
    self.runUntilCurrent()
--- <exception caught here> ---
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 797, in runUntilCurrent
    f(*a, **kw)
  File "/home/pouwelse/GIT-pulls/tribler/Tribler/dispersy/endpoint.py", line 250, in dispersythread_data_came_in
    u"standalone_ep")
  File "/home/pouwelse/GIT-pulls/tribler/Tribler/dispersy/dispersy.py", line 1462, in on_incoming_packets
    community.on_incoming_packets(list(iterator), cache, timestamp, source)
  File "/home/pouwelse/GIT-pulls/tribler/Tribler/dispersy/community.py", line 2023, in on_incoming_packets
    self._on_batch_cache(meta, batch)
  File "/home/pouwelse/GIT-pulls/tribler/Tribler/dispersy/community.py", line 2093, in _on_batch_cache
    self.on_messages(messages)
  File "/home/pouwelse/GIT-pulls/tribler/Tribler/dispersy/community.py", line 2207, in on_messages
    if self._dispersy.store_update_forward(possibly_messages, True, True, False):
  File "/home/pouwelse/GIT-pulls/tribler/Tribler/dispersy/dispersy.py", line 1698, in store_update_forward
    if self._update(possibly_messages) == False:
  File "/home/pouwelse/GIT-pulls/tribler/Tribler/dispersy/util.py", line 151, in wrapper
    return_value = func(*args, **kargs)
  File "/home/pouwelse/GIT-pulls/tribler/Tribler/dispersy/dispersy.py", line 1723, in _update
    messages[0].handle_callback(messages)
  File "/home/pouwelse/GIT-pulls/tribler/Tribler/dispersy/community.py", line 2881, in on_puncture
    candidate = self.create_or_update_walkcandidate(message.candidate.sock_addr, message.payload.source_lan_address, message.payload.source_wan_address, message.candidate.tunnel, u"unknown", message.candidate)
  File "/home/pouwelse/GIT-pulls/tribler/Tribler/dispersy/community.py", line 1495, in create_or_update_walkcandidate
    wcandidate.update(tunnel, lan_address, wan_address, connection_type)
  File "/home/pouwelse/GIT-pulls/tribler/Tribler/dispersy/candidate.py", line 300, in update
    assert lan_address == ("0.0.0.0", 0) or is_valid_address(lan_address), lan_address
exceptions.AssertionError: ('0.0.0.0', 7759)
```

**Comment by whirm** [comment]
@rjruigrok any idea about what's going on here?

**Comment by rjruigrok** [comment]
Not completely sure about this, currently `0.0.0.0:0` is used by the endpoint of a circuit to relay data back to the circuit originator. This has been working without problems for months, we should find out where it is passing port 7759 instead of 0.

**Comment by synctext** [comment]
btw 7759 is my default Tribler port.

**Comment by rjruigrok** [comment]
@whirm Might be the wan address voting on majority of received messages be the cause? It looks like this is the only place where 0.0.0.0 is set without a port number... 
https://github.com/Tribler/dispersy/blob/devel/dispersy.py#L800

**Comment by devos50** [comment]
@whirm I would assign someone to this issue who has experience with dispersy.

**Comment by whirm** [comment]
My guess is that the tunnel community is messing with dispersy.

**Comment by devos50** [comment]
After #2691, this one should not lead to a crash anymore but it's still unknown what caused this. I will postpone this issue to somewhere after Tribler 7.

**Comment by qstokkink** [comment]
It seems like the update conditional for the walk candidate updating is backwards.
[The lan](https://github.com/Tribler/dispersy/blob/devel/candidate.py#L305) and [the wan](https://github.com/Tribler/dispersy/blob/devel/candidate.py#L307) addresses should not be updated if their __current__ value is equal to `0.0.0.0:0`, __not__ if the new value being set is equal to `0.0.0.0:0`.

**Comment by devos50** [comment]
@qstokkink good find! Interested in fixing this one?

**Comment by qstokkink** [comment]
@devos50 sure, could you assign me please?
